types/ziggy-js/index.d.ts

```diff
@@ -0,0 +1,48 @@
+// Type definitions for ziggy-js 0.9
+// Project: https://github.com/tightenco/ziggy#readme
+// Definitions by: Ben Allfree <https://github.com/benallfree>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.6
+
+export type Param = string | number | string[] | number[] | undefined;
+
+export interface Params {
+    [key: string]: Param;
+}
+
+export interface Route {
+    uri: string;
+    methods: Array<'GET' | 'HEAD' | 'POST' | 'DELETE'>;
+    domain: null | string;
+}
```

Comment:
This object can also optionally have a `middleware` key with an array value and a `bindings` key with an object value where all the keys and values of that object are strings.

Reply:
I added them, but can you please point to where this is referenced in the ziggy js? I searched and didn't find those keys anywhere in the repo.

Reply:
I don't think `middleware` actually gets used by Ziggy's javascript anywhere, but it can be included with a config option, see https://github.com/tighten/ziggy/blob/0.9.x/src/RoutePayload.php#L98-L104 and https://github.com/tighten/ziggy/blob/0.9.x/tests/Unit/RoutePayloadTest.php#L280-L329.

`bindings` is going to be in `1.0` so you're right actually it's not possible yet, sorry!

Reply:
Hmm ok I think it is best to omit these for now, at least until 1.0 comes along. Typescript doesn't care if you give it extra keys, so it won't hurt anything if someone adds `{ middleware: 'foo' }` or even `{ bar: 'baz' }`

Does that sound ok to you? People can always include extra keys in TS, it will just treat them as `any` types.

Reply:
Perfect 👍